pkg/builder/iBuilder.go

```diff
@@ -54,12 +57,13 @@ type (
 	GoodnessPolarity int // -1 or 1
 	Threshold        float64
 	ScorecardCell    struct {
+		Mu               sync.Mutex
 		Data             DerivedDataElement
 		goodnessPolarity GoodnessPolarity
 		majorThreshold   Threshold
 		minorThreshold   Threshold
```

Comment:
Sounds good, thanks. I think you may have missed the mutex itself.

```suggestion
	ScorecardCell    struct {
		mu               sync.Mutex
		Data             DerivedDataElement
		goodnessPolarity GoodnessPolarity
		majorThreshold   Threshold
		minorThreshold   Threshold
```